amf-aml/shared/src/main/scala/amf/aml/internal/parse/common/AnnotationsParser.scala

```diff
@@ -1,85 +1,120 @@
+        val extension = DomainExtension()
+          .withId(id)
+          .set(DomainExtensionModel.Extension, parsedAnnotation, Annotations.inferred())
+          .withDefinedBy(property)
+          .withName(key)
+          .add(Annotations(value))
+        Some(extension)
+      case Failure(ex) =>
+        declarations.usedVocabs.get(ai.prefix.getOrElse("")) match {
+          case Some(vocabulary) =>
+            val id               = node.id + (if (node.id.endsWith("/") || node.id.endsWith("#")) "" else "/") + s"${ai.prefix.map(_ + "/").getOrElse("/")}$ai.suffix"
+            val parsedAnnotation = DynamicExtensionParser(value, Some(id))(ctx).parse()
+            val base             = vocabulary.base.value()
+            val propertyId       = if (base.endsWith("#") || base.endsWith("/")) base + ai.suffix else base + "/" + ai.suffix
+            val property         = CustomDomainProperty(Annotations(value)).withId(propertyId).withName(ai.key)
             val extension = DomainExtension()
               .withId(id)
               .set(DomainExtensionModel.Extension, parsedAnnotation, Annotations.inferred())
               .withDefinedBy(property)
-              .withName(k)
-              .add(Annotations(v))
+              .withName(ai.key)
+              .add(Annotations(value))
             Some(extension)
-          case Failure(ex) =>
-            declarations.usedVocabs.get(prefix.getOrElse("")) match {
-              case Some(vocabulary) =>
-                val id               = node.id + (if (node.id.endsWith("/") || node.id.endsWith("#")) "" else "/") + s"${prefix.map(_ + "/").getOrElse("/")}$suffix"
-                val parsedAnnotation = DynamicExtensionParser(v, Some(id)).parse()
-                val base             = vocabulary.base.value()
-                val propertyId       = if (base.endsWith("#") || base.endsWith("/")) base + suffix else base + "/" + suffix
-                val property         = CustomDomainProperty(Annotations(v)).withId(propertyId).withName(k)
-                val extension = DomainExtension()
-                  .withId(id)
-                  .set(DomainExtensionModel.Extension, parsedAnnotation, Annotations.inferred())
-                  .withDefinedBy(property)
-                  .withName(k)
-                  .add(Annotations(v))
-                Some(extension)
-              case None =>
-                ctx.eh.violation(MissingVocabulary, node.id, ex.getMessage, v)
-                None
-            }
+          case None =>
+            ctx.eh.violation(MissingVocabulary, node.id, ex.getMessage, value)
+            None
         }
-    } collect { case Some(parsed) => parsed }
+    }
+  }
 
-    if (parsedExtensions.nonEmpty) {
-      node.withCustomDomainProperties(parsedExtensions.toSeq)
+  private def parsedAnnotationProperties(ast: YMap): Seq[AnnotationInfo] = ast.entries.flatMap { entry: YMapEntry =>
```

Comment:
Probably needs ParserContext

Reply:
Seems that I erased it somewhere in the commit line so it should be ok.